lib/chef/resource/powershell_package_source.rb

```diff
@@ -0,0 +1,162 @@
+# Author:: Tor Magnus Rakvåg (tm@intility.no)
+# Copyright:: 2018, Intility AS
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class Resource
+    class PowershellPackageSource < Chef::Resource
+      resource_name "powershell_package_source"
+      provides(:powershell_package_source) { true }
+
+      description "Use the powershell_package_source resource to register a powershell package repository"
+      introduced "15.0"
+
+      property :name, String,
+               description: "The name of the package source",
+               name_property: true
+
+      property :url, String,
+               description: "The url to the package source",
+               required: true
+
+      property :trusted, [true, false],
+               description: "Whether or not to trust packages from this source",
+               default: false
+
+      property :provider_name, String,
+               equal_to: %w{ Programs msi NuGet msu PowerShellGet psl chocolatey },
+               validation_message: "The following providers are supported: 'Programs', 'msi', 'NuGet', 'msu', 'PowerShellGet', 'psl' or 'chocolatey'",
+               description: "",
+               default: "NuGet"
+
+      property :publish_location, String,
+               description: "The url where modules will be published to for this source. Only valid if the provider is 'PowerShellGet'.",
+               required: false
+
+      property :script_source_location, String,
+               description: "The url where scripts are located for this source. Only valid if the provider is 'PowerShellGet'.",
+               required: false
+
+      property :script_publish_location, String,
+               description: "The location where scripts will be published to for this source. Only valid if the provider is 'PowerShellGet'.",
+               required: false
+
+      load_current_value do
+        cmd = load_resource_state_script(name)
+        repo = powershell_out!(cmd)
```

Comment:
This looks like it all works, but for the future we've decided it's ok to require PowerShell 3.0 for all new resources. That gives you the ability to pipe output over to ConvertTo-Json for easy consumption within Chef.